Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -124,7 +124,15 @@ function getUpdateHistoryRequest(webAppPublishKuduUrl: string, isDeploymentSucce
 
     var collectionUrl = tl.getVariable('system.TeamFoundationCollectionUri'); 
     var teamProject = tl.getVariable('system.teamProject');
-
+	
+	var type = "Deployment";
+	var commitId = tl.getVariable('build.sourceVersion');
```

Comment:
You should handle the cases when the variable is not present. For all the variables.